backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Services/ResourceAdministrationPoint.cs

```diff
@@ -37,6 +40,26 @@ public class ResourceAdministrationPoint : IResourceAdministrationPoint
             _cacheConfig = cacheConfig.Value;
         }
 
+        /// <inheritdoc />
+        public async Task<PaginatedList<ServiceResourceFE>> GetPaginatedSearchResults(string languageCode, string[]? resourceOwnerFilters, string? searchString, int page, int resultsPerPage)
+        {
+            try
+            {
+                List<ServiceResource> resources = await GetFullResourceList();
+                List<ServiceResource> resourceList = resources.FindAll(r => r.ResourceType != ResourceType.MaskinportenSchema);
+                List<ServiceResourceFE> resourcesFE = MapResourceToFrontendModel(resourceList, languageCode);
+
+                List<ServiceResourceFE> filteredresources = FilterResourceList(resourcesFE, resourceOwnerFilters);
```

Comment:
But isn't that just the same number of operations anyway? 😅
The fact that they're separated into different loops shouldn't really matter, should it?

I mean, you get a performance of `O(n*f)` (where n is total num of resources and f is num of filters) for the filter-loop and then `O(m*s)`  (where m is num of resources and s is num of total search words) for the search, where f<<n, s<<m and m<=n, making the complete worst case scenario `O(n) + O(n)` which is `O(n)` anyway.

I guess the creation of extra variables is a thing, since you do store data in between function calls, but I don't know how much that affects performance in the grand scheme of things. I do think it is much more readable and reusable if we separate the operations though.

Reply:
Fair point, and yes the gains might be too minor to make it worth while over having an implementation which is easier to maintain, reuse and expand upon.

We'll anyway know more of how it performs once we get this to TT02/YT and can revisit both code and caching when we see if and where the shoe pinches.